src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +38,378 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
+    BgIkanaMirror_SetupWait(this);
+}
+
+void BgIkanaMirror_Destroy(Actor* thisx, PlayState* play) {
+    BgIkanaMirror* this = THIS;
+    s32 i;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(play, &this->mirrorCollider);
+
+    for (i = 0; i < ARRAY_COUNT(this->lightRaysColliders); i++) {
+        Collider_DestroyQuad(play, &this->lightRaysColliders[i]);
+    }
+}
+
+void BgIkanaMirror_SetupWait(BgIkanaMirror* this) {
+    this->isEmittingLight = 0;
+    this->actionFunc = BgIkanaMirror_Wait;
+}
+
+/*
+ * BgIkanaMirror_Wait is used to charge the mirror with light, increment the alpha value for the light absorption
+ * textures and decrement the alpha value for the light emission textures.
+ *
+ * By not directly setting the alpha values to 255 or 0, the transition from emitting light to
+ * absorbing it is visually smoother. Loading the proper display lists for absorption or emission of light is taken care
+ * of by BgIkanaMirror_Draw.
+ */
+void BgIkanaMirror_Wait(BgIkanaMirror* this, PlayState* play) {
+    s8 isEmittingLight;
+    s32 startEmittingLight = false;
+
+    // The light emission texture should gradually disappear from sight.
+    if (this->lightEmissionAlpha > 100) {
+        this->lightEmissionAlpha -= 100;
+    } else {
+        this->lightEmissionAlpha = 0;
+    }
+
+    // This checks if light is touching the mirror.
+    if (this->mirrorCollider.base.acFlags & AC_HIT) {
+        this->mirrorCollider.base.acFlags &= ~AC_HIT;
+        this->isEmittingLight = 0;
+
+        if (this->lightRayCharge < 400) {
+            this->lightRayCharge++;
+        }
+
+        if (this->lightAbsorptionAlpha < 195) {
+            this->lightAbsorptionAlpha += 60;
+        } else {
+            this->lightAbsorptionAlpha = 255;
+        }
+
+    } else {
+        isEmittingLight = this->isEmittingLight;
+
+        if (isEmittingLight > 0) {
+            startEmittingLight = true;
+        } else if (this->lightRayCharge > 0) {
+            this->isEmittingLight = isEmittingLight + 1;
+        }
+    }
+
+    if (startEmittingLight) {
+        BgIkanaMirror_SetupEmitLight(this);
+    } else {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->mirrorCollider.base);
+    }
+}
+
+void BgIkanaMirror_SetupEmitLight(BgIkanaMirror* this) {
+    this->dyna.actor.flags |= ACTOR_FLAG_20;
+    this->actionFunc = BgIkanaMirror_EmitLight;
+}
+
+/*
+ * BgIkanaMirror_Wait is used to deplete the mirror's light and release it, increment the alpha value for the light
+ * emission textures and decrement the alpha value for the light absorption textures.
+ *
+ * By not directly setting the alpha values to 0 or 255, the transition from absorbing light to
+ * releasing it is visually smoother. Loading the proper display lists for absorption or emission of light is taken care
+ * of by BgIkanaMirror_Draw.
+ */
```

Comment:
Same comment here, also had a typo
```suggestion
/**
 * Depletes the mirror's light and release it, increase alpha value for the light
 * emission textures and decrease the alpha value for the light absorption textures.
 */
```